### DISCUSSION TITLE
Branch protection rule does not work.

### DISCUSSION THREAD

**Discussion by Schwarzschild** [community/community#44000] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

I set up branch protection rules for the main branch of my public repo.  Mainly I want to prevent pushes directly to the main branch without a PR from another branch.  I did the Following.  I am still able to push commits directly to main.  Please advise.

<img width="906" alt="Screenshot 2023-01-11 at 2 37 43 PM" src="https://user-images.githubusercontent.com/3010072/211901661-a408668b-8f35-4f1a-a56c-4681cc99c4be.png">

**Comment by lejtemxviw** [comment]
I think you may want to also check the "require approval of the most recent push" checkbox - otherwise, maybe someone could approve a PR, then someone else could push additional changes into the branch and merge the branch with those additional changes, since GH would still recognize the PR as approved (from before the additional changes were put in).
This looks like our setup (though I cannot see it, myself :-p) - are you also an admin in that repo?  (I think these restrictions won't apply to you then).  Or is your main branch still called "master" rather than "main"?

**Comment by Schwarzschild** [comment]
Oh, thank you for the reply.  I will take your advice about recent push approval.  More importantly, I think you explained that because I am the owner the rule would not apply to me.  Funny thing is that I just starting this project and am the only developer.  I wanted the rule to apply to me too to keep me honest.

**Reply by lejtemxviw** [reply, marked as answer]
If you are the admin for that repo, I think there's a checkbox further down the page that allows to choose to apply these constraints even to admins...

**Comment by Schwarzschild** [comment]
You are right: I checked "Do not allow bypassing the above settings".

Thank you.
